tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,601 @@
+    assert konnected.YAML_CONFIGS not in hass.data[konnected.DOMAIN]
+
+
+async def test_setup_defined_hosts_known_auth(hass):
+    """Test we don't initiate a config entry if configured panel is known."""
+    MockConfigEntry(
+        domain="konnected",
+        unique_id="112233445566",
+        data={"host": "0.0.0.0", "id": "112233445566"},
+    ).add_to_hass(hass)
+    MockConfigEntry(
+        domain="konnected",
+        unique_id="aabbccddeeff",
+        data={"host": "1.2.3.4", "id": "aabbccddeeff"},
+    ).add_to_hass(hass)
+
+    assert (
+        await async_setup_component(
+            hass,
+            konnected.DOMAIN,
+            {
+                konnected.DOMAIN: {
+                    konnected.CONF_ACCESS_TOKEN: "abcdefgh",
+                    konnected.CONF_DEVICES: [
+                        {
+                            config_flow.CONF_ID: "aabbccddeeff",
+                            config_flow.CONF_HOST: "0.0.0.0",
+                            config_flow.CONF_PORT: 1234,
+                        },
+                    ],
+                }
+            },
+        )
+        is True
+    )
+
+    assert hass.data[konnected.DOMAIN][konnected.CONF_ACCESS_TOKEN] == "abcdefgh"
+    assert konnected.YAML_CONFIGS not in hass.data[konnected.DOMAIN]
+
+    # Flow aborted
+    assert len(hass.config_entries.flow.async_progress()) == 0
+
+
+async def test_setup_defined_hosts_no_known_auth(hass):
+    """Test we initiate config entry if config panel is not known."""
+    assert (
+        await async_setup_component(
+            hass,
+            konnected.DOMAIN,
+            {
+                konnected.DOMAIN: {
+                    konnected.CONF_ACCESS_TOKEN: "abcdefgh",
+                    konnected.CONF_DEVICES: [{konnected.CONF_ID: "aabbccddeeff"}],
+                }
+            },
+        )
+        is True
+    )
+
+    # Flow started for discovered bridge
+    assert len(hass.config_entries.flow.async_progress()) == 1
+
+
+async def test_config_passed_to_config_entry(hass):
+    """Test that configured options for a host are loaded via config entry."""
+    entry = MockConfigEntry(
+        domain=konnected.DOMAIN,
+        data={config_flow.CONF_ID: "aabbccddeeff", config_flow.CONF_HOST: "0.0.0.0"},
+    )
+    entry.add_to_hass(hass)
+    with patch.object(konnected, "AlarmPanel", autospec=True) as mock_int:
+        assert (
+            await async_setup_component(
+                hass,
+                konnected.DOMAIN,
+                {
+                    konnected.DOMAIN: {
+                        konnected.CONF_ACCESS_TOKEN: "abcdefgh",
+                        konnected.CONF_DEVICES: [{konnected.CONF_ID: "aabbccddeeff"}],
+                    }
+                },
+            )
+            is True
+        )
+
+    assert len(mock_int.mock_calls) == 3
+    p_hass, p_entry = mock_int.mock_calls[0][1]
+
+    assert p_hass is hass
+    assert p_entry is entry
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)